### PR TITLE
first pass renaming db->ledger

### PR DESCRIPTION
**Pull request by mschoch**
best effort was made to understand the context
and allow for "datbase" to continue to be used in it's more general context, "document database", "cloud database", etc.

also, no files were moved/renamed.
this was an intentional choice to avoid breaking links

this has one big downside, the category is still named 'Database API', and I would have changed that to 'Ledger API'
however, docusaurus is auto-generating the side-bar, and I couldn't find a way to separate the label shown from the URLs generated.  we can of course also simply update the broken links, if we are not concerned with breaking links outside, or seo.

### CODE REVIEW

**Comment by mschoch** [comment]
As discussed earlier, this pass is just db->ledger.
Further cleanup around connector/gateway will be separate PR (most likely stacked on this, assuming something close to this eventually gets merged)

File: docs/chatgpt-quick-start.md [hunk 1]
```diff
@@ -11,13 +11,13 @@ Fireproof is explicitly designed to be useful without requiring any setup or con
...
 ```
 
 For React you can also paste:
 
 ```jsx
-Fireproof/React/Usage: import { useLiveQuery, useDocument } from 'use-fireproof'; function App() { const result = useLiveQuery(doc => doc.word, { limit: 10 }); const [{ count }, setDoc, saveDoc] = useDocument({_id: 'count', count: 0}); return (<><p>{count} changes</p><input type='text' onChange={() => saveDoc({count: count + 1})} onSubmit={e => useLiveQuery.database.put({word: e.target.value})} /><ul>{result.map(row => (<li key={row.id}>{row.key}</li>))}</ul></>)}
+Fireproof/React/Usage: import { useLiveQuery, useDocument } from 'use-fireproof'; function App() { const result = useLiveQuery(doc => doc.word, { limit: 10 }); const [{ count }, setDoc, saveDoc] = useDocument({_id: 'count', count: 0}); return (<><p>{count} changes</p><input type='text' onChange={() => saveDoc({count: count + 1})} onSubmit={e => useLiveQuery.ledger.put({word: e.target.value})} /><ul>{result.map(row => (<li key={row.id}>{row.key}</li>))}</ul></>)}
```
Comment (jchris) [inline docs/chatgpt-quick-start.md:20]: I gotta think this one over and see it in the other repo

File: docs/concept-guide/scaling-fireproof.md [hunk 2]
```diff
@@ -4,31 +4,31 @@ sidebar_position: 5
...
 
-There are numerous compaction optimizations that can be implemented. Before 1.0, we need to decide on semantics for data preservation. Most databases allow for a system purge to remove old versions of data, either on a key or historic basis. Currently, Fireproof compaction preserves all database history, but discards unreferenced internal index blocks, etc. Each compaction links to the dataset it replaces, so it’s possible to reconstruct history from immutable archives even from a compacted lean dataset. This is a great topic to bring up in the [Discord](https://discord.gg/xC55bhgf6n).
+There are numerous compaction optimizations that can be implemented. Before 1.0, we need to decide on semantics for data preservation. Most ledgers allow for a system purge to remove old versions of data, either on a key or historic basis. Currently, Fireproof compaction preserves all ledger history, but discards unreferenced internal index blocks, etc. Each compaction links to the dataset it replaces, so it’s possible to reconstruct history from immutable archives even from a compacted lean dataset. This is a great topic to bring up in the [Discord](https://discord.gg/xC55bhgf6n).
```
Comment (jchris) [inline docs/concept-guide/scaling-fireproof.md:27]: first databases can stay

File: docs/connect.md [hunk 3]
```diff
@@ -50,8 +50,8 @@ Finally, connect to the party in your client code:
 import { useFireproof } from 'use-fireproof'
 import { connect } from '@fireproof/partykit'
 
-const { database } = useFireproof('my-app-database-name')
-const connection = connect.partykit(database, process.env.PUBLIC_PARTYKIT_HOST!)
+const { ledger } = useFireproof('my-app-ledger-name')
```
Comment (jchris) [inline docs/connect.md:53]: gotta check the react api here

File: docs/connect.md [hunk 4]
```diff
@@ -128,14 +128,14 @@ const inviteLinkCid = await cx.shareWith(myShareToken)
 The resulting CID looks something like this: `bafkreifzpfmjjmkrqwtzazzj46l6m4sa4umhyd5fv7hvjauyz2eftcvbda` -- this material is safe to publish because it can only be utilized by someone who has the private key the share token corresponds with. Send this `inviteLinkCid` to the person who's share token you used to initiate the share. They can then call:
 
 ```js
-const { database, connection } = await cx.joinShared(inviteLinkCid)
+const { ledger, connection } = await cx.joinShared(inviteLinkCid)
```
Comment (jchris) [inline docs/connect.md:131]: this is old

File: docs/concept-guide/scaling-fireproof.md (hunk 2 above)
Comment (jchris) [reply docs/concept-guide/scaling-fireproof.md:27]: the only fix that is blocking merge is this

File: docs/connect.md (hunk 4 above)
Comment (jchris) [reply docs/connect.md:131]: we should. leave it in for now and then do a cleanup

File: docs/concept-guide/scaling-fireproof.md (hunk 2 above)
Comment (mschoch) [reply docs/concept-guide/scaling-fireproof.md]: addressed in 0d1834

File: docs/connect.md (hunk 4 above)
Comment (mschoch) [reply docs/connect.md:131]: opened #12

File: docs/connect.md (hunk 3 above)
Comment (mschoch) [reply docs/connect.md:53]: marking this resolved, we have a cleanup of this file planed

**Pull request merged**